src/lib/arch/aarch64/Architecture.cc

```diff
@@ -88,24 +96,28 @@ Architecture::Architecture(kernel::Linux& kernel, YAML::Node config)
       }
     }
     // Store any opcode-based latency override
-    for (size_t j = 0; j < port_node["Instruction-Opcode"].size(); j++) {
-      uint16_t opcode = port_node["Instruction-Opcode"][j].as<uint16_t>();
+    for (size_t j = 0; j < port_node["Instruction-Opcodes"].num_children();
+         j++) {
+      uint16_t opcode;
+      port_node["Instruction-Opcodes"][j] >> opcode;
       opcodeExecutionInfo_[opcode].latency = latency;
       opcodeExecutionInfo_[opcode].stallCycles = throughput;
     }
   }
 
   // ports entries in the groupExecutionInfo_ entries only apply for models
   // using the outoforder core archetype
-  if (config["Core"]["Simulation-Mode"].as<std::string>() == "outoforder") {
+  if (config::SimInfo::getSimMode() == config::SimulationMode::Outoforder) {
     // Create mapping between instructions groups and the ports that support
     // them
-    for (size_t i = 0; i < config["Ports"].size(); i++) {
+    for (size_t i = 0; i < config["Ports"].num_children(); i++) {
       // Store which ports support which groups
-      YAML::Node group_node = config["Ports"][i]["Instruction-Group-Support"];
-      for (size_t j = 0; j < group_node.size(); j++) {
-        uint16_t group = group_node[j].as<uint16_t>();
-        uint8_t newPort = static_cast<uint8_t>(i);
+      ryml::ConstNodeRef group_node =
+          config["Ports"][i]["Instruction-Group-Support-Nums"];
+      for (size_t j = 0; j < group_node.num_children(); j++) {
+        uint16_t group;
+        group_node[j] >> group;
+        uint16_t newPort = static_cast<uint16_t>(i);
```

Comment:
Do we need to go from eight to 16 bits for newPort?

Reply:
Was changed as we've used more than 256 ports in simulations prior